source/melpomene/src/main.rs

```diff
@@ -153,77 +152,110 @@ fn kernel_entry(opts: MelpomeneOptions) {
 
     k.initialize(initialization_future).unwrap();
 
-    // Creating a dummy task to use the embedded display driver
-    let graphics_init_future = async move {
-        // Delay for one second, just for funsies
-        Delay::new(Duration::from_secs(1)).await;
+    // Create the interactive console task
+    let graphics_console = async move {
+        // Delay for 1.5 seconds, just for funsies
+        Delay::new(Duration::from_millis(1_500)).await;
 
-        EmbDisplay::register(k, 4, 160, 120).await.unwrap();
+        // Take Port 2 from the serial mux. This corresponds to TCP port 10002 when
+        // you are running crowtty
+        let mut mux_hdl = SerialMuxHandle::from_registry(k).await.unwrap();
+        let p2 = mux_hdl.open_port(2, 1024).await.unwrap();
+        drop(mux_hdl);
 
         let mut disp_hdl = EmbDisplayHandle::from_registry(k).await.unwrap();
-        let mut fc_0 = disp_hdl.get_framechunk(0, 80, 45, 160, 120).await.unwrap();
-        drop(disp_hdl);
-
-        let line_style = PrimitiveStyle::with_stroke(Gray8::WHITE, 1);
-        let tline = Line::new(Point::new(148, 2), Point::new(8, 2)).into_styled(line_style);
-        let bline = Line::new(Point::new(148, 24), Point::new(8, 24)).into_styled(line_style);
-        let lline = Line::new(Point::new(7, 24), Point::new(7, 2)).into_styled(line_style);
-        let rline = Line::new(Point::new(149, 24), Point::new(149, 2)).into_styled(line_style);
-        let text_style = MonoTextStyle::new(&FONT_6X9, Gray8::WHITE);
-        let datetime_style = MonoTextStyle::new(&FONT_5X7, Gray8::WHITE);
-        let text1 = Text::new("Welcome to mnemOS!", Point::new(10, 10), text_style);
-        let text2 = Text::new("A tiny operating system", Point::new(10, 20), text_style);
-        let output_settings = OutputSettingsBuilder::new()
-            .theme(BinaryColorTheme::OledBlue)
-            .build();
-
-        let mut sdisp = SimulatorDisplay::<Gray8>::new(Size::new(320, 240));
-        let mut window = Window::new("mnemOS", &output_settings);
+        let char_y = PROFONT_12_POINT.character_size.height;
+        let char_x = PROFONT_12_POINT.character_size.width + PROFONT_12_POINT.character_spacing;
+
+        // Draw titlebar
```

Comment:
not a blocker, just a thought for the future: it seems like a bunch of this stuff ought to move out of melpomene and into the platform agnostic kernel crate? it would be nice if the kernel could get a display driver handle from the driver registry and draw the shell and stuff, instead of doing it in melpomene. 

i guess we would probably want to be doing stuff like determining what font size to use based on the display resolution instead of hard coding it, so it might be a bit more complicated. none of this is a blocker for merging this change, just a note for the future.

Reply:
Yeah, this is probably worth a think about how to split this up. Right now the only impl is in melpomene, and there's no real "standard init process": the kernel is more like a grab bag of lib components that you need to glue together.

I think we should look at this again when we start moving bits from the simulator into the kernel itself so we can reuse them on the D1 hardware.